File: scheduling/scheduling-annotation-processor/src/test/java/ru/tinkoff/kora/scheduling/annotation/processor/controller/ScheduledJdkAtFixedRateTest.java
package ru.tinkoff.kora.scheduling.annotation.processor.controller;

import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleAtFixedRate;

import java.time.temporal.ChronoUnit;

public class ScheduledJdkAtFixedRateTest {
    @ScheduleAtFixedRate(initialDelay = 100, period = 1000, config = "baseline", unit = ChronoUnit.MILLIS)
    public void baseline() {

    }

    @ScheduleAtFixedRate(initialDelay = 100, period = 1000, unit = ChronoUnit.MILLIS)
    public void noConfig() {

    }

    @ScheduleAtFixedRate(config = "onlyConfig")
    public void onlyConfig() {

    }

    @ScheduleAtFixedRate(period = 1000)
    public void onlyRequired() {

    }

    @ScheduleAtFixedRate(period = 1000, config = "onlyRequiredWithConfig")
    public void onlyRequiredWithConfig() {

    }
}

File: scheduling/scheduling-ksp/src/test/kotlin/ru/tinkoff/kora/scheduling/ksp/controller/ScheduledJdkAtFixedRateTest.kt
package ru.tinkoff.kora.scheduling.ksp.controller

import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleAtFixedRate
import java.time.temporal.ChronoUnit

class ScheduledJdkAtFixedRateTest {
    @ScheduleAtFixedRate(initialDelay = 100, period = 1000, config = "baseline", unit = ChronoUnit.MILLIS)
    fun baseline() {
    }

    @ScheduleAtFixedRate(initialDelay = 100, period = 1000, unit = ChronoUnit.MILLIS)
    fun noConfig() {
    }

    @ScheduleAtFixedRate(config = "onlyConfig")
    fun onlyConfig() {
    }

    @ScheduleAtFixedRate(period = 1000)
    fun onlyRequired() {
    }

    @ScheduleAtFixedRate(period = 1000, config = "onlyRequiredWithConfig")
    fun onlyRequiredWithConfig() {
    }
}
